server/models/Stakeholder.js

```diff
@@ -516,76 +261,64 @@ const deleteRelation = (repo, current_id) => async (stakeholder) => {
       removeObj.child_id = type === 'children' ? data.id : id;
     }
 
-    const stakeholderRelation = await repo.deleteRelation(removeObj);
+    const stakeholderRelation =
+      await this._stakeholderRepository.deleteRelation(removeObj);
 
     return stakeholderRelation;
   }
-  throw new Error(
-    "Whoops! That stakeholder link can't be updated, no permission",
-  );
-};
-
-const updateStakeholder =
-  (repo, owner_id = null) =>
-  async (data) => {
-    const orgId = Number(owner_id);
-    // get organization from old entity table
-    const id =
-      owner_id === null || Number.isNaN(orgId)
-        ? owner_id
-        : await getUUID(repo, orgId);
-
-    const editedStakeholder = StakeholderTree({ ...data });
-    const relatedStakeholders = await repo.getRelatedIds(id);
-    const foundStakeholder = await repo.verifyById(editedStakeholder.id);
-
-    // confirm they have right to edit
-    if (
-      (id && relatedStakeholders.includes(editedStakeholder.id)) ||
-      foundStakeholder.owner_id === id ||
-      id === null
-    ) {
-      // remove children and parents temporarily to update
-      const { children, parents, ...updateObj } = editedStakeholder;
-      const stakeholder = await repo.updateStakeholder(updateObj);
-
-      return StakeholderTree({ ...stakeholder, children, parents });
-    }
-    throw new Error({
-      message: "Whoops! That stakeholder can't be edited, no permission",
-    });
-  };
 
-const createStakeholder =
-  (repo, owner_id = null) =>
-  async (newStakeholder) => {
-    const orgId = Number(owner_id);
-    // get organization from old entity table
-    const id =
-      owner_id === null || Number.isNaN(orgId)
-        ? owner_id
-        : await getUUID(repo, orgId);
+  async updateStakeholder(data) {
```

Comment:
We do need to pass along the id here, `updateStakeholder(id, data)`

Reply:
Noted, although the id is not currently used in the function

Reply:
You're right, but without it the response was breaking.  If we remove it then I think it needs to be removed in the service and handler too.